components/elm/src/cpl/lnd_disagg_forc.F90

```diff
@@ -0,0 +1,797 @@
+    real(r8) :: tmp_Snow_frc        ! Current snow fraction
+
+    real(r8) :: e
+    real(r8) :: qvsat
+
+    real(r8) :: sum_qbot_g    ! weighted sum of column-level lwrad
+    real(r8) :: sum_wtsq_g      ! sum of weights that contribute to sum_lwrad_g
+    real(r8) :: qbot_norm_g   ! normalization factors
+    real(r8) :: sum_lwrad_g    ! weighted sum of column-level lwrad
+    real(r8) :: sum_wtslw_g      ! sum of weights that contribute to sum_lwrad_g
+    real(r8) :: lwrad_norm_g   ! normalization factors
+    real(r8) :: esatw                ! saturation vapor pressure over water (Pa)
+    real(r8) :: esati                ! saturation vapor pressure over ice (Pa)
+    real(r8) :: a0,a1,a2,a3,a4,a5,a6 ! coefficients for esat over water
+    real(r8) :: b0,b1,b2,b3,b4,b5,b6 ! coefficients for esat over ice
+    real(r8) :: tdc, temp            ! Kelvins to Celcius function and its input
+    real(r8) :: vp                   ! water vapor pressure (Pa)
+    
+    real(r8), allocatable :: deltaRain(:)      ! Deviation of subgrid rain from grid rain
+    real(r8), allocatable :: deltaSnow(:)      ! Deviation of subgrid snow from grid snow
+    real(r8) :: deltaR              ! Temporary deltaRain
+    real(r8) :: deltaS              ! Temporary deltaSnow
+    real(r8) :: sum_of_hrise        ! Sum of height rise of air parcel of all subgrids of a grid
+    real(r8) :: hrise               ! Temporary height rise
+    real(r8) :: elvrnge             ! Elevation range between lowest and highest tpu
+    real(r8) :: ave_elv
+    integer :: elv_flag             ! Elevation flag to trac grids with +ve grid elevation and -ve tpu elevation
+
+    integer :: uaflag = 0
+    integer :: precip_dwn = 0         ! Used to turn on/off the downscaling of precipitation  0 = on; 1 = off
+    integer :: other_forcing_dwn = 0  ! Used to turn on/off the downscaling of other forcing  0 = on; 1 = off
+
+    character(len=*), parameter :: subname = 'downscale_grd_to_topounit'
+    !----------------------------------------------------------------------------------------
+    
+    ! Constants to compute vapor pressure
+    parameter (a0=6.107799961_r8    , a1=4.436518521e-01_r8, &
+         a2=1.428945805e-02_r8, a3=2.650648471e-04_r8, &
+         a4=3.031240396e-06_r8, a5=2.034080948e-08_r8, &
+         a6=6.136820929e-11_r8)
+
+    parameter (b0=6.109177956_r8    , b1=5.034698970e-01_r8, &
+         b2=1.886013408e-02_r8, b3=4.176223716e-04_r8, &
+         b4=5.824720280e-06_r8, b5=4.838803174e-08_r8, &
+         b6=1.838826904e-10_r8)
+
+    !
+    ! function declarations
+    !
+    tdc(temp) = min( 50._r8, max(-50._r8,(temp-SHR_CONST_TKFRZ)) )
+    esatw(temp) = 100._r8*(a0+temp*(a1+temp*(a2+temp*(a3+temp*(a4+temp*(a5+temp*a6))))))
+    esati(temp) = 100._r8*(b0+temp*(b1+temp*(b2+temp*(b3+temp*(b4+temp*(b5+temp*b6))))))
+    !-----------------------------------------------------------------------
```

Comment:
Add a comment here that includes the citation for these saturated vapor pressure expressions for water and ice.

Reply:
@thorntonpe after discussion with Michael, I have added comments that these expressions are taken from lnd_import_export.F90.